sql/src/main/java/io/crate/planner/statement/CopyStatementPlanner.java

```diff
@@ -68,124 +67,150 @@ public CopyStatementPlanner(ClusterService clusterService) {
-            } else {
-                symbol = reference;
-            }
-            toCollect.add(symbol);
-        }
-        // add clusteredBy column (if not part of primaryKey)
-        if (clusteredByPrimaryKeyIdx == -1 && table.clusteredBy() != null &&
-            !DocSysColumns.ID.equals(table.clusteredBy())) {
-            toCollect.add(table.getReference(table.clusteredBy()));
-        }
-        // add _raw or _doc
-        if (table.isPartitioned() && analysis.partitionIdent() == null) {
-            toCollect.add(table.getReference(DocSysColumns.DOC));
-        } else {
-            toCollect.add(table.getReference(DocSysColumns.RAW));
-        }
-
-        // add columns referenced by generated columns which are used as partitioned by column
-        for (Reference reference : referencedReferences) {
-            if (!toCollect.contains(reference)) {
-                toCollect.add(reference);
-            }
-        }
+        // if there are partitionValues (we've had a PARTITION clause in the statement)
+        // we need to use the calculated partition values because the partition columns are likely NOT in the data being read
+        // the partitionedBy-inputColumns created for the projection are still valid because the positions are not changed
+        rewriteToCollectToUsePartitionValues(table.partitionedByColumns(), partitionValues, toCollect);
 
         DiscoveryNodes allNodes = clusterService.state().nodes();
         FileUriCollectPhase collectPhase = new FileUriCollectPhase(
             context.jobId(),
             context.nextExecutionPhaseId(),
             "copyFrom",
-            getExecutionNodes(allNodes, analysis.settings().getAsInt("num_readers", allNodes.getSize()), analysis.nodePredicate()),
-            analysis.uri(),
+            getExecutionNodes(allNodes, copyFrom.settings().getAsInt("num_readers", allNodes.getSize()), copyFrom.nodePredicate()),
+            copyFrom.uri(),
             toCollect,
             projections,
-            analysis.settings().get("compression", null),
-            analysis.settings().getAsBoolean("shared", null)
+            copyFrom.settings().get("compression", null),
+            copyFrom.settings().getAsBoolean("shared", null)
         );
 
         Collect collect = new Collect(collectPhase, TopN.NO_LIMIT, 0, 1, 1, null);
         return Merge.ensureOnHandler(collect, context, Collections.singletonList(MergeCountProjection.INSTANCE));
     }
 
+    private static void rewriteToCollectToUsePartitionValues(List<Reference> partitionedByColumns,
+                                                             List<BytesRef> partitionValues,
+                                                             List<Symbol> toCollect) {
+        for (int i = 0; i < partitionValues.size(); i++) {
+            Reference partitionedByColumn = partitionedByColumns.get(i);
+            int idx = toCollect.indexOf(partitionedByColumn);
```

Comment:
probably need to do a GeneratedReference check here and change the indexOf call to use the generatedExpression